### DISCUSSION TITLE
Is "Exceeds maximum number of requests waiting to be resolved, please try again" message related to 429 responses?

### DISCUSSION THREAD

**Discussion by mmednik-noves** [MetaMask/snaps#2492] [Q&A]
Hey guys!

Is Metamask returning this message for **429 responses**?:
`Exceeds maximum number of requests waiting to be resolved, please try again.`

We haven't found a pattern to reproduce it yet. That's why we believe it is associated with a 429 response from our API. But we wanted to be sure in what scenarios Metamask displays that message.

Our extension (https://snaps.metamask.io/snap/npm/noves-foresight/) performs a transaction foresight, making a request to our API.
For 99% of the transactions we tested, everything works great!

Only once or twice we seen that error message. And **we actually have a max requests limit for our API.**

Thanks!

**Comment by mmednik-noves** [comment]
Those transactions with that message were on the Ethereum mainnet.

**Comment by FlaviuWeb3** [comment]
Hi @mmednik-noves My hunch is that the response is coming from your API, being rate limited, though I will chase internally to see how we return the 429 response.

**Comment by FrederikBolding** [comment]
@mmednik-noves This happens when the Snap has multiple requests waiting to be processed but the Snap hasn't been started yet.

Do you have a way to reliably reproduce this? It should definitely not happen often 🤔
